auto_update.go

```diff
@@ -0,0 +1,134 @@
+package main
+
+import (
+	"archive/zip"
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/blang/semver"
+
+	"github.com/inconshreveable/go-update"
+)
+
+// RunUpdater checks if an update is due, checks if current version is outdated and performs update if needed
+func RunUpdater(app *TGFApplication) bool {
+	var autoUpdateFile = "tgfautoupdate"
+	var dueForUpdate = lastRefresh(autoUpdateFile) > 2*time.Hour
+	if !dueForUpdate {
+		app.Debug("Update not due")
+		return false
+	}
+	touchImageRefresh(autoUpdateFile)
+
+	v, err := getLatestVersion()
+	if err != nil {
+		app.Debug("Unable to fetch latest version from S3", err)
+		return false
+	}
+
+	latestVersion, err := semver.Make(v)
+	if err != nil {
+		app.Debug("Semver error", err)
+		return false
+	}
+
+	currentVersion, err := semver.Make(version)
+	if err != nil {
+		app.Debug("Semver error", err)
+		return false
+	}
+
+	if !currentVersion.LT(latestVersion) {
```

Comment:
I don't like negatives:
```suggestion
	if currentVersion.GE(latestVersion) {
```

Reply:
You know negation of a negative negation is still negative 😋.